### DISCUSSION TITLE
Get params without hook

### DISCUSSION THREAD

**Discussion by motionrus** [TanStack/router#2138] [General]
can i get params without hook

<img width="1338" alt="image" src="https://github.com/user-attachments/assets/b6c08109-89bf-4998-8421-8361d0e82681">

**Comment by ViewableGravy** [comment]
You can access the router state either with the hook `useRouterState` or by accessing `__state` on the router. You should be able to access the route params from here if you need them outside of the router/react.

**Reply by motionrus** [reply]
@ViewableGravy `useRouterState` is a hook, I can't use hooks outside the component, can you give me an example with `__state` please

**Reply by ViewableGravy** [reply]
Sorry about that, I am aware that useRouterState is a hook, but I meant to say that you could then pass the router through to a function in react land. Of course that is useless since you could've just called useParams then so sort of a null point.

As for accessing params from state. The way I would do this is the following:
```ts
const params = router.__state.state.matches.at(-1).params
```

The last match should be your most specific match, and this contains an object (params) that contains all of the params which is the same as useParams.
